addons/auth_signup/models/res_partner.py

```diff
@@ -162,36 +125,60 @@ def _signup_retrieve_partner(self, token, check_validity=False, raise_exception=
             :param raise_exception: if True, raise exception instead of returning False
             :return: partner (browse record) or False (if raise_exception is False)
         """
-        self.env.cr.execute("SELECT id FROM res_partner WHERE signup_token = %s AND active", (token,))
-        partner_id = self.env.cr.fetchone()
-        partner = self.browse(partner_id[0]) if partner_id else None
+        partner = self._get_partner_from_token(token)
         if not partner:
-            if raise_exception:
-                raise exceptions.UserError(_("Signup token '%s' is not valid", token))
-            return False
-        if check_validity and not partner.signup_valid:
-            if raise_exception:
-                raise exceptions.UserError(_("Signup token '%s' is no longer valid", token))
-            return False
+            raise exceptions.UserError(_("Signup token '%s' is no longer valid or expired", token))
         return partner
 
     @api.model
-    def signup_retrieve_info(self, token):
+    def _signup_retrieve_info(self, token):
         """ retrieve the user info about the token
             :return: a dictionary with the user information:
-                - 'db': the name of the database
                 - 'token': the token, if token is valid
-                - 'name': the name of the partner, if token is valid
-                - 'login': the user login, if the user already exists
-                - 'email': the partner email, if the user does not exist
```

Comment:
It looks like the method is still returning all of these, so this documentation can remain, except that now it returns None if the token is invalid, and otherwise all of these.